src/libstore/local-fs-store.cc

```diff
@@ -79,8 +79,6 @@ ref<FSAccessor> LocalFSStore::getFSAccessor()
 
 void LocalFSStore::narFromPath(const StorePath & path, Sink & sink)
 {
-    if (!isValidPath(path))
-        throw Error("path '%s' is not valid", printStorePath(path));
```

Comment:
If the issue is that build.cc is calling `Store::makeContentAddressed()` on an invalid path, then maybe that function should take a separate argument for the path (so it can use `dumpPath()`).